lib/main.js

```diff
@@ -0,0 +1,147 @@
+const oracledb = require('oracledb')
+const Store = require('jsreport-sql-store')
+
+module.exports = async (reporter, definition) => {
+  if (reporter.options.store.provider !== 'oracle') {
+    definition.options.enabled = false
+    return
+  }
+
+  function transformBindVarsBoolean (bindVars) {
+    for (let index = 0; index < bindVars.length; index++) {
+      const bindVar = bindVars[index]
+
+      if (bindVar === true) {
+        bindVars[index] = 1
+      } else if (bindVar === false) {
+        bindVars[index] = 0
+      }
+    }
+  }
+
+  function transformResultBoolean (opts, res) {
+    function tranformResultBooleanForType (type) {
+      if (type) {
+        for (let index = 0; index < res.rows.length; index++) {
+          const row = res.rows[index]
+
+          for (const property in row) {
+            const propDef = type[property]
+
+            if (propDef && propDef.type === 'Edm.Boolean') {
+              if (row[property] === 1) {
+                row[property] = true
+              } else if (row[property] === 0) {
+                row[property] = false
+              }
+            }
+          }
+        }
+      }
+    }
+
+    if (res.rows && opts.entitySet) {
+      const entitySet = reporter.documentStore.model.entitySets[opts.entitySet]
+
+      if (entitySet) {
+        const entityType = entitySet.entityType.substring(9) // remove "jsreport." prefix
+        tranformResultBooleanForType(reporter.documentStore.model.complexTypes[entityType])
+        tranformResultBooleanForType(reporter.documentStore.model.entityTypes[entityType])
+      }
+    }
+  }
+
+  async function executeQuery (q, opts = {}) {
+    async function execute (conn) {
+      const bindVars = []
+
+      for (let i = 0; i < q.values.length; i++) {
+        bindVars.push(q.values[i])
+      }
+
+      transformBindVarsBoolean(bindVars)
+
+      const res = await conn.execute(q.text, bindVars, { outFormat: oracledb.OUT_FORMAT_OBJECT })
+
+      transformResultBoolean(opts, res)
+
+      return {
+        records: res.rows,
+        rowsAffected: res.rowsAffected
+      }
+    }
+
+    let conn
+
+    if (!opts.transaction) {
```

Comment:
I think you should always release the connection in the finally block to avoid leaks when commit or execution fails.

```js
try {
  if (!opts.transaction) {
      conn = await pool.getConnection()
  } else {
     conn = opts.transaction
  }

   const res = await execute(conn)

  if (!opts.transaction) {
        await conn.commit()    
  } 
}
finally {
  if (conn) {
      await conn.close()
  }
}
```

Reply:
ok try/finally is better, I don't think it should be closed when its an transaction and no exception is thrown?

Reply:
Ups. Yes true, The closing for transactions is handled in `transactionManager ` manager part. Also the rollback gets called automatically on error which closes.

So I guess it should be like this here.
```js
finally {
  if (conn && !opts.transaction) {
      await conn.close()
  }
}
```